permission/client/client.go

```diff
@@ -46,3 +46,41 @@ func (c *Client) GetUserPermissions(ctx context.Context, requestUserID string, t
 
 	return permission.FixOwnerPermissions(result), nil
 }
+
+func (c *Client) HasMembershipRelationship(ctx context.Context, granteeUserID, grantorUserID string) (has bool, err error) {
+	fromTo, err := c.GetUserPermissions(ctx, granteeUserID, grantorUserID)
+	if err != nil {
+		return false, err
+	}
+	if len(fromTo) > 0 {
+		return true, nil
+	}
+	toFrom, err := c.GetUserPermissions(ctx, grantorUserID, granteeUserID)
+	if err != nil {
+		return false, err
+	}
+	if len(toFrom) > 0 {
+		return true, nil
+	}
+	return false, nil
+}
+
+func (c *Client) HasCustodianPermissions(ctx context.Context, granteeUserID, grantorUserID string) (has bool, err error) {
+	perms, err := c.GetUserPermissions(ctx, granteeUserID, grantorUserID)
+	if err != nil {
+		return false, err
+	}
+	return len(perms[permission.Custodian]) > 0, nil
+}
+
+func (c *Client) HasWritePermissions(ctx context.Context, granteeUserID, grantorUserID string) (has bool, err error) {
+	if granteeUserID != "" && granteeUserID == grantorUserID {
+		return true, nil
+	}
+	perms, err := c.GetUserPermissions(ctx, granteeUserID, grantorUserID)
+	if err != nil {
+		return false, err
+	}
+	return len(perms[permission.Custodian]) > 0 || len(perms[permission.Write]) > 0 || len(perms[permission.Owner]) > 0, nil
```

Comment:
If gatekeeper returns:
```
{
   custodian: {}
}
```

We will deserialize this to `map[string]Permission` where `Permission` is `map[string]interface{}`.  If I am not mistaken `len(perms[permission.Custodian])` check whether that second map is empty, which will always be true, no?

Reply:
Yes I think I only handled the case where the user created his own account.